src/core/server/elasticsearch/client/scoped_cluster_client.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ClientWrapper } from './client_wrapper';
+import { ClientFacade } from './client_facade';
+
+/** @public **/
+export interface IScopedClusterClient {
+  asInternalUser: () => ClientFacade;
+  asCurrentUser: () => ClientFacade;
+}
+
+/** @internal **/
+export class ScopedClusterClient implements IScopedClusterClient {
+  constructor(
+    private readonly internalClient: ClientWrapper,
+    private readonly scopedClient: ClientWrapper
+  ) {}
+
+  asInternalUser() {
+    return this.internalClient;
```

Comment:
The `ScopedClusterClient` now have even less logic that the legacy class, and is just a container for the internal and (preconfigured with headers) scoped clients.